### PR TITLE
Order by relation count / various

### LINKED ISSUE #520: Document supported date formats
SQLite is very forgiving about datetimes; it actually doesn't have a real datetime type at all!

So what it supports then is either a string, or an integer. The integer is easy, it's just seconds from 1.1.1970 midnight, but the string is more complex.

Right now we support only the standard datetime strings:

- [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) (`1996-12-19T16:39:57-08:00`)
- [RFC 2822](https://tools.ietf.org/html/rfc2822#section-3.3) (`Tue, 1 Jul 2003 10:52:37 +0200`)

So if one is using an existing database in SQLite, the stringified dates need to follow one of these formats, or exist as integers.

### LINKED ISSUE #782: Document `prisma format`
Source for content: https://github.com/prisma/specs/tree/master/cli#format

### LINKED ISSUE #1238: Setting transaction isolation level
https://prisma-company.slack.com/archives/CEYCG2MCN/p1611854206072700

### LINKED ISSUE #1345: Relation aggregates
Related: https://github.com/prisma/prisma/issues/5079

Internal context: https://prisma-company.slack.com/archives/CFM8D5VHC/p1615798173004300

### LINKED ISSUE #1400: Parallel requests warning / connectOrCreate
https://prisma-company.slack.com/archives/CEYCG2MCN/p1615293931008500?thread_ts=1615288922.004700&cid=CEYCG2MCN

### PR DESCRIPTION
**Pull request by mhwelander**

### CODE REVIEW

File: content/200-concepts/100-components/02-prisma-client/050-filtering-and-sorting.mdx [hunk 1]
```diff
@@ -310,6 +310,32 @@ const posts = await prisma.post.findMany({
 })
 ```
 
+### Sort by relation aggregate (Preview)
+
```
Comment (mhwelander) [inline content/200-concepts/100-components/02-prisma-client/050-filtering-and-sorting.mdx:319]:
```suggestion
<div class="videoWrapper">

<iframe width="560" height="315" src="https://www.youtube.com/embed/vmVUeZny73o" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

</div>
```

File: content/200-concepts/100-components/02-prisma-client/050-filtering-and-sorting.mdx [hunk 2]
```diff
@@ -310,6 +310,36 @@ const posts = await prisma.post.findMany({
 })
 ```
 
+### Sort by relation aggregate (Preview)
+<div class="videoWrapper">
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/vmVUeZny73o" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+</div>
```
Comment (mhwelander) [inline content/200-concepts/100-components/02-prisma-client/050-filtering-and-sorting.mdx:318]:
```suggestion
</div>

```

File: content/400-reference/200-api-reference/050-prisma-client-reference.mdx [hunk 3]
```diff
@@ -2566,6 +2607,80 @@ const user = await prisma.user.update({
...
+  | :--------------------------------------------------------------- | :--------------------------------------------------------------- |
+  | Query hits server, starts transaction A                          | Query hits server, starts transaction B                          |
+  |                                                                  | Find record where `tagName` equals `computing`, record not found |
+  | Find record where `tagName` equals `computing`, record not found |                                                                  |
+  |                                                                  | Create record where `tagName` equals `computing` and connect     |
+  | Create record where `tagName` equals `computing`                 |                                                                  |
+  | Unique violation, record already created by transaction B        |                                                                  |
+
+  To work around this scenario, we recommed catching the unique violation exception (`PrismaClientKnownRequestError`, error `P2002`) and retrying failed queries.
```
Comment (matthewmueller) [inline content/400-reference/200-api-reference/050-prisma-client-reference.mdx]: recommend

File: content/400-reference/200-api-reference/200-command-reference.mdx [hunk 4]
```diff
@@ -487,6 +489,68 @@ model Profile {
...
+
+```
+Environment variables loaded from prisma\.env
+Prisma schema loaded from prisma\schema.prisma
+Error: Schema parsing
+error: The preview feature "orderByRelationnp" is not known. Expected one of: microsoftSqlServer, groupBy, createMany, orderByRelation, napi
+  -->  schema.prisma:3
+   | 
+ 2 |   provider        = "prisma-client-js"
+ 3 |   previewFeatures = ["orderByRelationnp"]
```
Comment (nikolasburk) [inline content/400-reference/200-api-reference/200-command-reference.mdx]:
```suggestion
 3 |   previewFeatures = ["orderByRelation"]
```

File: content/400-reference/200-api-reference/050-prisma-client-reference.mdx (hunk 3 above)
Comment (janpio) [reply content/400-reference/200-api-reference/050-prisma-client-reference.mdx]:
```suggestion
  To work around this scenario, we recommend catching the unique violation exception (`PrismaClientKnownRequestError`, error `P2002`) and retrying failed queries.
```

**Pull request merged**